vignettes/design-principles.Rmd

```diff
@@ -28,10 +28,19 @@ The main rationale is to allow downstream analysis and modelling packages to sea
 ### `contactmatrix`
 
 - [x] Inherits from array (matrix for 2D) to leverage existing methods, and utility functions (e.g., `t()`)
-- [x] Puts survey participants ("from") in rows
+    - ✓ used in socialmixr
+    - ✓ used in contactdata
+    - ✓ used in conmat
+- [x] Puts survey participants ("from") in rows:
+    - ✓ used in socialmixr
+    - ✓ used in contactdata
+    - ✗ transposed in conmat
 - [x] Levels are ordered lexicographically (using C locale)
 - [x] Boolean `attr(cm, “symmetric”)` to determine if $c_{ij}N_i == c_{ji}N_j$
-- [ ] Standardized notation to denote age-bins
+- [ ] Standardized notation to denote age-bins (`[low, high)`):
```

Comment:
Maybe it should be a follow up in the socialmixr but doesn't this mean that socialmixr is not entirely consistent then? It use `[,)` for `contact_matrix()` output, but not for `limits_to_agegroups()`.

Reply:
Yes, correct. We also provide https://epiforecasts.io/socialmixr/reference/reduce_agegroups.html but it might be better to do all operations in the `[,)` notation. So the workflow from age to groups is

``` r
library("socialmixr")
#> 
#> Attaching package: 'socialmixr'
#> The following object is masked from 'package:utils':
#> 
#>     cite
seq_len(20) |>
  reduce_agegroups(limits = c(0, 5, 10)) |>
  limits_to_agegroups()
#>  [1] 0-4 0-4 0-4 0-4 5-9 5-9 5-9 5-9 5-9 10+ 10+ 10+ 10+ 10+ 10+ 10+ 10+ 10+ 10+
#> [20] 10+
#> Levels: 0-4 < 5-9 < 10+
```

<sup>Created on 2024-05-03 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>